virtcontainers/container.go

```diff
@@ -1142,6 +1142,21 @@ func (c *Container) detachDevices() error {
 		}
 	}
 
+	for _, dev := range c.devices {
+		if err := c.sandbox.devManager.RemoveDevice(dev.ID); err != nil {
```

Comment:
Thanks for addressing the first comment here, but you didn't answer/addressed the second one:
> is there any reason why we cannot call this `RemoveDevice()` function right after we detached the device with `DetachDevice()` ? This would prevent from running twice through the same list of devices.

Reply:
Nice suggestion, I think this could work.
Will modify